linkml/generators/pythongen.py

```diff
@@ -841,7 +949,13 @@ def gen_postinit(self, cls: ClassDefinition, slot: SlotDefinition) -> Optional[s
                     or slot.range in self.schema.types
                     or slot.range in self.schema.enums
                 ):
-                    rlines.append(f"\tself.{aliased_slot_name} = {base_type_name}(self.{aliased_slot_name})")
+                    enum_forward_ref = self.ifabsent_processor.enum_forward_references
+                    if cls.name in enum_forward_ref.keys() and slot.range in enum_forward_ref[cls.name].keys():
+                        rlines.append(
```

Comment:
sort of having a hard time telling what's happening here, but i'll see if running it and inspecting the output clears things up

Reply:
This is probably the trickiest part of the whole patch. It is probably worth a more extensive comment explaining it.

Reply:
From what I understand, in the case of a forward reference for an enum it will return something like:

```python
self.my_slot = MyType.MyEnum(MyPermissibleValue)
```

And in the case of a basic type field, it will return:

```python
self.my_slot = MyType(self.my_slot)
```

Did I understand that correctly ?

Reply:
@vincentkelleher you are right, it should be `MyEnum.MyPermissibleValue`, I will check if it returns it or something wrong (like you seem to expect from my code).

Reply:
The code was in fact buggy, but not as @vincentkelleher thought.

And this confirms that @sneakers-the-rat was right to "complain" about the lack of a specific test
> i can't see a test for the underlying problem (an ifabsent value that is a value of an enum) here

My poor-man regression test `test_ifabsent_stuff` was only asserting that the module could be generated and compiled, not that the resulting code was the expected one... I am almost done with the bug fixing and the corresponding test for `ifabsent` on enumerations.

Reply:
Now `test_ifabsent_stuff` also tests the correctness of the `ifabsent` in the case of enumerations.